### PR TITLE
fix(UI): remove the ButtonSpacer

### LINKED ISSUE #48759: Remove the buttonSpacer used to add margin between components
**Describe the bug**
We have a component called buttonSpacer and its propose is to add margin between components. We can remove the need for it and use CSS to add the margin.

https://github.com/Sboonny/freeCodeCamp/blob/1f0070630a101d87a6f3b483b8cdb3173dbb155d/client/src/components/helpers/button-spacer.tsx#L1-L10?plain=1

Usually, I try to think of a reason for it, but the only thing that I can come up with is, this is a relic of the past, and we don't need it anymore.

If there are special cases in which `margin` needs to be added, CSS selectors can handle that. And if it's there a need for consistency between spacing, adding a gap to its parent component should do the trick, as well.

---

Here is how it looks without the buttonSpacer, here is the page https://www.freecodecamp.org/settings

With the buttonSpacer

image

Without the buttonSpacer

image

### PR DESCRIPTION
**Pull request by mmatsumoto1026**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #48759

This PR was opened to remove the `ButtonSpacer` and replace it to modified `Spacer` helper.

### CODE REVIEW

**Review by Sboonny — COMMENTED** [review]
nicely done👍

File: client/src/components/helpers/spacer.tsx [hunk 1]
```diff
@@ -4,19 +4,11 @@ interface SpacerProps {
...
   ) : (
-    <>
-      {Array.from(Array(size), (_, i) => (
-        <Comp key={`spacer_${i}`} />
-      ))}
-    </>
+    <div className='spacer' style={{ padding: `${size}px 0`, height: '1px' }} />
   );
 
 export default Spacer;
```
[the remark this answers is not in the document]
Comment (mmatsumoto1026) [reply client/src/components/helpers/spacer.tsx:14]: Now I realized that there is no `size === 0` situation in the production code so I agree with removing the condition.

Actually, there are `<Spacer />` for `padding: 15px 0;` in the code, so I will push the suggested code after adding `size={15}` prop to the `<Spacer />`.

[the remark this answers is not in the document]
Comment (Sboonny) [reply client/src/components/helpers/spacer.tsx:14]: Oh the existing one is removed in another pr, sorting the conflicts should make your life easier.

You are talking about the next to the home page image right?🤔

[the remark this answers is not in the document]
Comment (mmatsumoto1026) [reply client/src/components/helpers/spacer.tsx:14]: Thanks, I resolved the conflicts.

And yes, the `<Spacer />` which is next to `CampersImage`. But also other ones in the codebase.

**Review by Sboonny — APPROVED** [review]
LGTM 🎉, thank you.

**Review by moT01 — COMMENTED** [review]
I would be fine adding this. The only concern is that it maybe gives a little too much flexibility the spacer padding - someone could give it a padding of 11px or something. We could potentially give the spacer a few standard sizes to use - maybe `small` = 5px, `medium` = 15px, `large` = 30px.

What do you think about that @mmatsumoto1026 @Sboonny? It's pretty minor, I would be fine with it as-is if we want.

**Comment by Sboonny** [comment]
> We could potentially give the spacer a few standard sizes to use - maybe small = 5px, medium = 15px, large = 30px.

This sounds good to me, but I think it should be done in a second pr, because spacer is used in different prs and the longer this isn't merged, the more old styled spacers will be added

**Comment by mmatsumoto1026** [comment]
Thanks for your review, @moT01.

I agree with your idea that gives the spacer a few standard sizes (it might also be `ex-large` = 45px). 

In this PR, I only focused to remove `ButtonSpacer` and modify the `Spacer` helper so it can use instead the one.

And I can try to open the following PR to improve the code to optimize the spacing flexibility regarding spacer size and make spacing consistent.

**Comment by ahmadabdolsaheb** [comment]
Thanks for taking care of this @mmatsumoto1026, I agree with Tom assessment. 
We would preferably need just a few even sizes, but that is for another pr.

**Review by ahmaxed — APPROVED** [review]

**Pull request merged**